cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/stream/FileStreamAdmin.java

```diff
@@ -433,11 +433,12 @@ public StreamConfig create(final StreamId streamId, @Nullable final Properties p
       specifiedOwnerPrincipal == null ? null : new KerberosPrincipalId(specifiedOwnerPrincipal);
     if (!streamId.getNamespaceId().equals(NamespaceId.SYSTEM) && principalId == null) {
       // if stream owner is not present, get the namespace impersonation principal
-      String namespacePrincipal = ownerAdmin.getOwnerPrincipal(streamId.getNamespaceId());
+      String namespacePrincipal = ownerAdmin.getImpersonationPrincipal(streamId.getNamespaceId());
```

Comment:
This is because https://issues.cask.co/browse/CDAP-8176, we currently do not have owner of namespace stored in the owner store. `getOwnerPrincipal` will only check the owner store, and we will always get null. `getImpersonationPrincipal` will check the entity is NamespaceId, and return the correct owner.